contracts/Cruize.sol

```diff
@@ -0,0 +1,130 @@
+ * @title Cruize AssetPool.
+ */
+contract Cruize is Proxy, AaveV2Wrapper {
+
+    /** @dev function receive will allow contract to direct receive ether 
+     from other address. ex -  WETH contract and other's wallet. 
+     */
+    receive() external payable {
+        // if someone accidentally sent ETH to our contract
+        // then we will deposit that amount to our protocal
+        // and it will prevent eth to mis place in SC.
+        if (msg.sender != WETH) {
+            // depositAsset(msg.value, WETH);
+        }
+    }
+
+    //----------------------------//
+    //     Mutation Functions     //
+    //----------------------------//
+
+    /**
+     * @notice initialize will initialize the crContract.
+     * @param _cruize_wallet address -  Cruize Wallet address.
+     * @param _crContract address -  ERC20Upgradeable Contract.
+     */
+
+    function initialize(address _cruize_wallet, address _crContract)
+        external
+        initializer
+    {
+        __Ownable_init();
+        __ReentrancyGuard_init();
+        require(_crContract != address(0));
+        require(_cruize_wallet != address(0));
+        crContract = _crContract;
+        cruize_wallet = _cruize_wallet;
+    }
+
+    /**
+     * @notice createToken  will Clone CRTokenUpgradeable (ERC20 token).
+     * @param name name of   ERC20Upgradeable Contract.
+     * @param symbol symbol of   ERC20Upgradeable Contract.
+     * @param decimal decimal value of ERC20Upgradeable Contract.
+     * @dev incase of ETH  , we will take token address of WETH becuase
+     * ETH is a native token of Ethereum so we don't have token address for ETH.
+     */
+
+    function createToken(
+        string memory name,
+        string memory symbol,
+        address reserve,
+        address tokenOracle,
+        uint8 decimal
+    ) external onlyOwner nonReentrant {
+        if (lpTokens[reserve] != address(0)) revert AssetAlreadyExists();
+        if (tokenOracle == address(0) || reserve == address(0))
+            revert ZeroAddress();
+
+        ILPtoken crToken = ILPtoken(createClone(crContract));
+        lpTokens[reserve] = address(crToken);
+        oracles[reserve] = tokenOracle;
+        crToken.initialize(name, symbol, decimal);
+        emit CreateToken(address(crToken), name, symbol, decimal);
+    }
+
+    /**
+     * @notice  deposit will deposit user's asset to aave lending pool
+     * and take loan of about 25% of collateral amount.
+     * @param amount number of token to be deposit.
+     * @param reserve token address to deposit.
+     */
+    function deposit(uint256 amount, address reserve)
+        external
+        payable
+        nonReentrant
+    {
+        uint256 borrowAmount = depositToAave(reserve, amount);
+        borrow(reserve, borrowAmount); // borrow from aave and transfer them to cruize wallet
+    }
+
+    /**
+     * @dev Cruize pool will repay the debt amount
+     */
+    function repay(uint256 amount) public nonReentrant {
```

Comment:
it should be an internal  function rather than public it's mean anyone can this repay function

Reply:
also there must be some restrictions  on call this functions

Reply:
No, it should be the public function because we have to repay debt in different intervals, and no will dare to repay our loan because he/she will not get anything from it.